### PR TITLE
Modify database directory 

### PR DESCRIPTION
**Pull request by LoicCadour**
Current behaviour : 
when you selected a dir, it was moving the db in this dir: cut and paste.

Now,
when you select a dir:
if this dir is empty, it creates a brand new database.
if this dir is not empty AND has a db file in it, it simply points on this db.
in other cases, the change is not applied.

I also managed the case where the db path does not exist anymore (if it were on an external hard disk that has been ejected) for reading and writing on the db.

### CODE REVIEW

**Comment by mathildemerle** [comment]
perfect under linux \o/

File: src/medCore/gui/settingsWidgets/medDatabaseSettingsWidget.cpp [hunk 1]
```diff
@@ -33,40 +33,46 @@ medDatabaseSettingsWidget::medDatabaseSettingsWidget(QWidget *parent) :
...
-     }
+    QDir defaultPath = QDir(medStorage::dataLocation());
+    defaultPath.cdUp();
+
+    QString path = QFileDialog::getExistingDirectory(this, tr("Select the database directory"),
+                                                     QDir(defaultPath).absolutePath(),
+                                                     QFileDialog::ShowDirsOnly
+                                                     | QFileDialog::DontResolveSymlinks);
+
+    if(path != "") //the user hasn't clicked on Cancel
```
Comment (mathildemerle) [inline src/medCore/gui/settingsWidgets/medDatabaseSettingsWidget.cpp]: "did not click" no? :)

Comment (LoicCadour) [inline src/medCore/gui/settingsWidgets/medDatabaseSettingsWidget.cpp]: I always suck at preterit vs present perfect :)

File: src/medCore/gui/settingsWidgets/medDatabaseSettingsWidget.cpp [hunk 2]
```diff
@@ -33,40 +33,46 @@ medDatabaseSettingsWidget::medDatabaseSettingsWidget(QWidget *parent) :
...
+    {
+        if (!QDir(path).entryInfoList(QDir::NoDotAndDotDot | QDir::Dirs | QDir::Files).isEmpty() && QDir(path).entryInfoList(QStringList("db")).isEmpty())
+        {
+            QMessageBox::information( this, tr("Database location not valid"),
+                                      tr("The directory selected is not valid:\nEither it doesn't point to an existing medInria database directory \nOR your new directory is not empty. \nPlease choose another one."));
```
Comment (mathildemerle) [inline src/medCore/gui/settingsWidgets/medDatabaseSettingsWidget.cpp]: medInria->MUSIC i think :)

Comment (mathildemerle) [inline src/medCore/gui/settingsWidgets/medDatabaseSettingsWidget.cpp]: Or maybe: "The directory selected is not valid:\nEither it doesn't point to an existing database directory \nOR your new directory is not empty. \nPlease choose another one."
No need of the software name :)

File: src/medCore/gui/settingsWidgets/medDatabaseSettingsWidget.cpp (hunk 1 above)
Comment (fcollot) [inline src/medCore/gui/settingsWidgets/medDatabaseSettingsWidget.cpp]: both are correct

File: src/medCore/gui/settingsWidgets/medDatabaseSettingsWidget.cpp (hunk 2 above)
Comment (fcollot) [inline src/medCore/gui/settingsWidgets/medDatabaseSettingsWidget.cpp]: Maybe there could be a compiler variable for the software name to make our lives simpler.

Comment (fcollot) [inline src/medCore/gui/settingsWidgets/medDatabaseSettingsWidget.cpp]: (not suggesting for this PR of course, just an idea)

**Comment by LoicCadour** [comment]
It also works on Windows :)

**Comment by LoicCadour** [comment]
Please review this PR ASAP.
I just destroyed my db trying to switch db with the current master code.

**Comment by fcollot** [comment]
approved

**Pull request merged**